third_party/blink/renderer/core/inspector/inspector_network_agent.cc

```diff
@@ -1260,6 +1278,23 @@ void InspectorNetworkAgent::PrepareRequest(DocumentLoader* loader,
     }
     request.SetDevToolsAcceptedStreamTypes(std::move(accepted_stream_types));
   }
+
+  // Capture the record replay bookmark for the network request here,
+  // where the devtools stack id is taken.
+  if (recordreplay::IsRecordingOrReplaying()) {
+    std::string url_string = request.Url().GetString().Utf8().c_str();
```

Comment:
Can you make sure to include the ScriptForbiddenScope::AllowUserAgentScript here, as in https://github.com/replayio/chromium/pull/141 ?

Reply:
Just for that one, or for all of them?  If I can allow user-agent script arbitrarily at these event points, I can change this implementation to not have the awkward temp C++ vector storing events until pulled in by JS.

Reply:
I'm not sure what you mean by "all of them", as the NewBookmark call here is the one which is problematic and there is only one of those calls in this file.  I think you can allow user-agent scripts to run here if that simplifies things, see also https://linear.app/replay/issue/RUN-633

Reply:
I was just looking at the code above instead of below, and inferring that you were saying we wanted this at all the sites where we pipe events.

Understood now.